app/2.1.x/go.md

```diff
@@ -4,44 +4,128 @@ title: Go language support
+The [go-pdk][go-pdk] allows your Go plugins to access functionality provided
+by the [Kong PDK][kong-pdk]. In order to use it, you need to import add
+`"github.com/Kong/go-pdk"` to the list of packages imported by your plugin.
+The `kong` parameter received on the event handler methods is the entry point
+for the Go PDK functions. Most of these functions work the same as the corresponding
+functions in the Lua PDK.
 
-## Development
+See the [go-pdk](https://godoc.org/github.com/Kong/go-pdk) godoc page for the
+reference documentation of the Go PDK
 
-To write a Kong plugin in Go, you need to:
+### Prerequisites
+
+In order to use Go plugins, you need the following:
+
+1. A Go compiler, which can be obtained from your Operating System's package
+manager, or directly at the [Go Downloads page](https://golang.org/dl/);
+
+2. The `go-pluginserver` executable. By default, Kong will expect it in
+`/usr/local/bin`; however, this location can be customized with the `go_pluginserver_exe`
+property in the Kong configuration file - or via environment variables, like
+any other Kong configuration property;
+
+3. Set the `go_plugins_dir` variable in the Kong configuration file to a valid directory
+where your Go plugins will be installed; the default value `off` value disables
+Go support;
+
+4. Add your compiled Go plugins to the directory specified in the previous step
+(`go_plugins_dir`);
+
+5. As with custom Lua plugins, make sure to add your Go plugin name to the `plugins`
+property, which effectively means Kong will load your plugin;
+
+6. Enable the plugin in the usual way, either via Admin API or a declarative
+configuration file.
 
-* Define a structure type to hold configuration.
-* Write a `New()` function to create instances of your structure.
-* Add methods on that structure to handle events.
-* Compile like: `go build -buildmode=plugin`.
-* Put the resulting library (the `.so` file) into the `go_plugins_dir` directory.
+### Building the Go Plugin Server and your Go Plugins
 
+The Go Plugin Server is a typical Go application. Leveraging Go Modules, it
+can be built in the following way:
+
+1. Initialize a `go.mod` file for your Kong Go Plugin project:
+```
+$ go mod init kong-go-plugin
+go: creating new go.mod: module kong-go-plugin
+```
+
+2. Import the Go Plugin Server as a dependency:
+```
+$ go get -d -v github.com/Kong/go-pluginserver
+go: github.com/Kong/go-pluginserver upgrade => v0.5.0
+go: downloading github.com/Kong/go-pluginserver v0.5.0
+go: downloading github.com/Kong/go-pdk v0.5.0
+go: downloading github.com/ugorji/go v1.1.7
+go: downloading github.com/ugorji/go/codec v1.1.7
+```
+
+This will, by default, select the latest version available, registering it
+into the `go.mod` file and adding all its dependencies into `go.sum`. Note that,
+in this example, the latest go-pluginserver version available was `v0.5.0` and
+it also pinned the go-pdk version it uses.
+
+3. The Go Plugin Server can be built with the following command, which
```

Comment:
i feel that using "...can be built..." sounds like there's an option, but it's the only way.  maybe something a bit more imperative, like "3. Build the Go Plugin Server.  The following command results in the executable placed in the current directory.  Move it to where Kong expects it (`/usr/local/bin` by default)"